### ISSUE TITLE
Styled Components Fails build in Gatsby 3.2

### ISSUE THREAD

**Issue by ashhitch** [gatsbyjs/gatsby#30591]
## Description

When running a build an exception is thrown:
```
window is not available during server-side rendering.
WebpackError: ReferenceError: window is not defined (from plugin: gatsby-plugin-styled-components)
```

### Steps to reproduce
Update dependencies to Gatsby 3.2 
Run Gatsby build (on Gatsby cloud)

### Expected result

Build to complete and plugin to take care of the window references in Styled Components

### Actual result

Exception is thrown

### Environment

  System:
    OS: macOS 11.1
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.8.0 - ~/.nvm/versions/node/v14.8.0/bin/node
    Yarn: 1.0.2 - /usr/local/bin/yarn
    npm: 6.14.7 - ~/.nvm/versions/node/v14.8.0/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.90
    Edge: 89.0.774.63
    Safari: 14.0.2
  npmPackages:
    gatsby: ^3.2.0 => 3.1.0 
    gatsby-cli: ^3.2.0 => 3.1.0 
    gatsby-image: ^3.2.0 => 3.1.0 
    gatsby-omni-font-loader: ^1.3.0 => 1.3.0 
    gatsby-plugin-catch-links: ^3.2.0 => 3.1.0 
    gatsby-plugin-create-client-paths: ^3.2.0 => 3.1.0 
    gatsby-plugin-google-analytics: ^3.2.0 => 3.1.0 
    gatsby-plugin-google-fonts: ^1.0.1 => 1.0.1 
    gatsby-plugin-image: ^1.2.0 => 1.2.0 
    gatsby-plugin-loadable-components-ssr: ^2.1.0 => 2.1.0 
    gatsby-plugin-manifest: ^3.2.0 => 3.1.0 
    gatsby-plugin-netlify: ^3.2.0 => 3.1.0 
    gatsby-plugin-offline: ^4.2.0 => 4.1.0 
    gatsby-plugin-react-helmet: ^4.2.0 => 4.1.0 
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0 
    gatsby-plugin-recaptcha: ^1.0.5 => 1.0.5 
    gatsby-plugin-robots-txt: ^1.5.5 => 1.5.5 
    gatsby-plugin-sass: ^4.2.0 => 4.1.0 
    gatsby-plugin-sharp: ^3.2.0 => 3.1.0 
    gatsby-plugin-sitemap: ^3.2.0 => 3.1.0 
    gatsby-plugin-styled-components: ^4.1.0 => 4.1.0 
    gatsby-plugin-typescript: ^3.2.0 => 3.1.0 
    gatsby-plugin-web-font-loader: ^1.0.4 => 1.0.4 
    gatsby-plugin-wpgraphql-seo: ^1.1.4 => 1.1.4 
    gatsby-source-filesystem: ^3.2.0 => 3.1.0 
    gatsby-source-wordpress: ^5.2.0 => 5.1.0 
    gatsby-transformer-sharp: ^3.2.0 => 3.1.0 
  npmGlobalPackages:
    gatsby-cli: 2.18.0

**Comment by herecydev** [comment]
There's loads of bugs being raised against styled components, like this one: https://github.com/styled-components/styled-components/issues/3444

Basically just stay away from version `5.2.2` and you'll be fine

**Comment by krismorf** [comment]
Fixed in `5.2.3`

**Comment by ashhitch** [comment]
will close this as its an issue with Styled-components, not the gatsby plugin:

reverting back down to 5.2.1 fixes the issue, while we wait for 5.2.3
```
  "resolutions": {
    "styled-components": "5.2.1"
  }
```
